src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/VsHierarchyExtensions.cs

```diff
@@ -27,31 +27,59 @@ public static Guid GetGuidProperty(this IVsHierarchy hierarchy, VsHierarchyPropI
         }
 
         /// <summary>
-        ///     Gets the value of the specified property if the hierarchy supports it.
+        ///     Gets the value of the specified property if the hierarchy supports it, or throws an excepton if there was an error.
         /// </summary>
         public static T GetProperty<T>(this IVsHierarchy hierarchy, VsHierarchyPropID property, T defaultValue = default(T))
         {
             return GetProperty(hierarchy, HierarchyId.Root, property, defaultValue);
         }
 
         /// <summary>
-        ///     Gets the value of the specified property if the hierarchy supports it.
+        ///     Gets the value of the specified property of the specified item if the hierarchy supports it, or throws an exception if there was an error.
         /// </summary>
         public static T GetProperty<T>(this IVsHierarchy hierarchy, HierarchyId item, VsHierarchyPropID property, T defaultValue = default(T))
         {
+            HResult hr = GetProperty(hierarchy, item, property, defaultValue, out T result);
```

Comment:
I didn't change behavior - see the HRESULT returning one below. We were always throwing unless it was DISP_E_MEMBERNOTFOUND.